prepare/libreswan.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+LIBRESWAN_DIR="/opt/ipsec"
+
+
+# Allow only users with root priviledge to run the script.
+if [ $EUID -ne 0 ]; then
+    echo "This script can be run with root priviledge only." >&2
+    exit 1
+fi
+
+# Check for the correct # of arguments - TODO
+
+
+libreswan_gen_connection ()
+{
+	CONNECTION_CFG="$1"
+	MODE="$2"
+
+	echo "conn roadwarrior_psk
+	auto=add
+	authby=secret
+	pfs=no
+	rekey=no
+	left=172.31.70.1
+	leftsubnet=0.0.0.0/0
+	rightaddresspool=172.29.100.2-10
+	right=%any
+	cisco-unity=yes
+	leftxauthserver=yes
+	rightxauthclient=yes
+	leftmodecfgserver=yes
+	rightmodecfgclient=yes
+	modecfgpull=yes
+	xauthby=alwaysok
+	ike-frag=yes
+	ikev2=never" > "$CONNECTION_CFG"
+	if [ "$MODE" = "aggressive" ]; then
+		echo \
+"	rightid=@yolo
+	aggressive=yes" > "$CONNECTION_CFG"
+	fi
+}
+
+libreswan_gen_secrets ()
+{
+	SECRETS_CFG="$1"
+
+	echo ": PSK \"ipsecret\"" > "$SECRETS_CFG"
+	chmod 600 "$SECRETS_CFG"
+} 
+
+libreswan_gen_netconfig ()
+{
+	ip netns add libreswan
+
+	# IPv6 on a veth confuses pluto. Sigh. (TODO: check id still true)
+	# ERROR: bind() for 80/80 fe80::94bf:8cff:fe1b:7620:500 in process_raw_ifaces(). Errno 22: Invalid argument
+	echo 1 > /proc/sys/net/ipv6/conf/default/disable_ipv6
+	ip netns exec libreswan echo 1 > /proc/sys/net/ipv6/conf/default/disable_ipv6
+	ip link add libreswan0 type veth peer name libreswan1
+	ip link set libreswan0 netns libreswan
+
+	ip netns exec libreswan ip link set lo up
+	ip netns exec libreswan ip addr add dev libreswan0 172.31.70.1/24
+	ip netns exec libreswan ip link set libreswan0 up
+	ip link set dev libreswan1 up
+
+	ip netns exec libreswan dnsmasq --pid-file=/tmp/libreswan_dnsmasq.pid \
+	                                --dhcp-range=172.31.70.2,172.31.70.40,2m \
+	                                --interface=libreswan0 --bind-interfaces
+	sleep 5
+	echo "PID of dnsmasq:"
+	pidof dnsmasq
```

Comment:
if you write a pid-file, you should read that instead.

Reply:
addressed